vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -803,6 +808,13 @@ void AdsImpl::OnDidServeAdNotification(const AdNotificationInfo& ad) {
 void AdsImpl::OnAdNotificationViewed(const AdNotificationInfo& ad) {
   account_->Deposit(ad.creative_instance_id, ad.type,
                     ConfirmationType::kViewed);
+
+  auto impression_served_at_covariate_log_entry =
+      std::make_unique<ImpressionServedAtCovariateLogEntry>();
+  impression_served_at_covariate_log_entry->SetLastImpressionTimestamp(
+      base::Time::Now());
+  CovariateLogs::Get()->SetCovariateLogEntry(
+      std::move(impression_served_at_covariate_log_entry));
```

Comment:
Can we decouple this code to a util, something along the lines of `ad_notification_viewed_covariate_log_traning_util`

Reply:
Summarised in in last comment, closing this one